File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/ApplicationActAdm/ApplicationActMainFAdm.java
package com.example.adminapp.ApplicationActAdm;

import android.os.Bundle;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Adapter;
import android.widget.FrameLayout;

import com.example.adminapp.GroupFragAdm.GrpQuizesAdm.QuizesMainScAdm;
import com.example.adminapp.R;
import com.google.android.material.floatingactionbutton.FloatingActionButton;

import java.util.ArrayList;


public class ApplicationActMainFAdm extends Fragment {

    private RecyclerView recyclerVApplicationActMainFAdm;
    private FrameLayout frameLayApplicationActMainFAdm;
    private FloatingActionButton floatingBtnApplicationActMainFAdm;

    private ArrayList<DataModelRecVApplicationMainFAdm> dataHolder;
    private AdapterApplicationMainFAdm adapter;

    public ApplicationActMainFAdm() {
        // Required empty public constructor
    }


    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = LayoutInflater.from(getContext()).inflate(R.layout.fragment_application_act_main_f_adm, container, false);

        recyclerVApplicationActMainFAdm = view.findViewById(R.id.recyclerVApplicationActMainFAdm);
        frameLayApplicationActMainFAdm = view.findViewById(R.id.frameLayApplicationActMainFAdm);
        floatingBtnApplicationActMainFAdm = view.findViewById(R.id.floatingBtnApplicationActMainFAdm);
        floatingBtnApplicationActMainFAdm.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Fragment fragment = new Custom_Add_ApplicationOptAdm();
                FragmentTransaction fragmentTransaction = getFragmentManager().beginTransaction();
                fragmentTransaction.replace(R.id.frameLayApplicationActAdm, fragment).commit();
                fragmentTransaction.addToBackStack(null);
            }
        });

        recyclerVApplicationActMainFAdm.setLayoutManager(new LinearLayoutManager(getContext()));
        addDataToHolder();
        recyclerVApplicationActMainFAdm.setAdapter(new AdapterApplicationMainFAdm(dataHolder));


        return view;
    }

    private void addDataToHolder() {
        dataHolder = new ArrayList<>();
        DataModelRecVApplicationMainFAdm obj = new DataModelRecVApplicationMainFAdm(
                "#58464",
                "Leave Application",
                "Demo Description here"
        );
        dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);
        dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);dataHolder.add(obj);
    }
}

File: KnowledgeSpaceApk/adminapp/src/main/java/com/example/adminapp/ApplicationActAdm/ApplicationActMainFAdm.kt
package com.example.adminapp.ApplicationActAdm

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.FrameLayout
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.adminapp.R
import com.google.android.material.floatingactionbutton.FloatingActionButton

class ApplicationActMainFAdm : Fragment() {
    private var recyclerVApplicationActMainFAdm: RecyclerView? = null
    private var frameLayApplicationActMainFAdm: FrameLayout? = null
    private var floatingBtnApplicationActMainFAdm: FloatingActionButton? = null
    private var dataHolder: ArrayList<DataModelRecVApplicationMainFAdm?>? = null
    private val adapter: AdapterApplicationMainFAdm? = null
    override fun onCreateView(
        inflater: LayoutInflater, container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        val view = LayoutInflater.from(context)
            .inflate(R.layout.fragment_application_act_main_f_adm, container, false)
        recyclerVApplicationActMainFAdm = view.findViewById(R.id.recyclerVApplicationActMainFAdm)
        frameLayApplicationActMainFAdm = view.findViewById(R.id.frameLayApplicationActMainFAdm)
        floatingBtnApplicationActMainFAdm =
            view.findViewById(R.id.floatingBtnApplicationActMainFAdm)
        floatingBtnApplicationActMainFAdm?.setOnClickListener(View.OnClickListener {
            val fragment: Fragment = Custom_Add_ApplicationOptAdm()
            val fragmentTransaction = requireFragmentManager().beginTransaction()
            fragmentTransaction.replace(R.id.frameLayApplicationActAdm, fragment).commit()
            fragmentTransaction.addToBackStack(null)
        })
        recyclerVApplicationActMainFAdm?.setLayoutManager(LinearLayoutManager(context))
        addDataToHolder()
        recyclerVApplicationActMainFAdm?.setAdapter(AdapterApplicationMainFAdm(dataHolder))
        return view
    }

    private fun addDataToHolder() {
        dataHolder = ArrayList()
        val obj = DataModelRecVApplicationMainFAdm(
            "#58464",
            "Leave Application",
            "Demo Description here"
        )
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
        dataHolder!!.add(obj)
    }
}